socrates/lib/middleware/accessrights.js

```diff
@@ -1,55 +1,66 @@
 'use strict';
 var conf = require('simple-configure');
-var Member = conf.get('beans').get('member');
+var beans = conf.get('beans');
+var Member = beans.get('member');
+var socratesActivitiesService = beans.get('socratesActivitiesService');
 
 module.exports = function accessrights(req, res, next) {
-  res.locals.accessrights = {
-    req: req,
+  socratesActivitiesService.getCurrentSocrates(function (err, socratesActivity) {
+    var socrates = err ? null : socratesActivity;
+    res.locals.accessrights = {
+      req: req,
 
-    member: function () {
-      return this.req.user && this.req.user.member;
-    },
+      member: function () {
+        return this.req.user && this.req.user.member;
+      },
 
-    isRegistered: function () {
-      return !!this.member();
-    },
+      isRegistered: function () {
+        return !!this.member();
+      },
 
-    memberId: function () {
-      return this.isRegistered() ? this.member().id() : null;
-    },
+      memberId: function () {
+        return this.isRegistered() ? this.member().id() : null;
+      },
 
-    isMember: function (member) {
-      return this.isRegistered() && this.memberId() === member.id();
-    },
+      isMember: function (member) {
+        return this.isRegistered() && this.memberId() === member.id();
+      },
 
-    isSuperuser: function () {
-      return Member.isSuperuser(this.memberId()); // same superusers as in SWK
-    },
+      isSuperuser: function () {
+        return Member.isSuperuser(this.memberId()); // same superusers as in SWK
+      },
 
-    isSoCraTesAdmin: function () {
-      return this.isRegistered() && conf.get('socratesAdmins').indexOf(this.memberId()) > -1;
-    },
+      isSoCraTesAdmin: function () {
+        return this.isRegistered() && conf.get('socratesAdmins').indexOf(this.memberId()) > -1;
+      },
 
-    canEditMember: function (member) {
-      return this.isMember(member);
-    },
+      canEditMember: function (member) {
+        return this.isMember(member);
+      },
 
-    canDeleteMember: function (member) {
-      return this.isSuperuser() && !this.isMember(member);
-    },
+      canDeleteMember: function (member) {
+        return this.isSuperuser() && !this.isMember(member);
+      },
 
-    canCreateActivity: function () {
-      return this.isSuperuser();
-    },
+      canCreateActivity: function () {
+        return this.isSuperuser();
+      },
 
-    canEditActivity: function () {
-      return this.isSuperuser() || this.isSoCraTesAdmin();
-    },
+      canEditActivity: function () {
+        return this.isSuperuser() || this.isSoCraTesAdmin();
+      },
 
-    canDeleteActivity: function () {
-      return this.isSuperuser();
-    }
+      canDeleteActivity: function () {
+        return this.isSuperuser();
+      },
+
+      needsToPay: function () {
+        return !!socrates && socrates.resources().resourceNamesOf(this.memberId()).length > 0
```

Comment:
Ich denke, die beiden boolean coercions in L. 58 und 59 sind unnötig, das && macht das schon.